### PR TITLE
test: install software-properties-common if absent during PPA setup

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message
```
test: install software-properties-common if absent during PPA setup

In minimal images, add-apt-repository command will be absent because
software-properties-common is not installed.

This breaks integration tests which use ImageType.MINIMAL and attempt
to install cloud-init from a PPA.

Before calling add-apt-repository, check presence of the command.
When absent, apt install software-properties-common.
```

## Additional Context

https://github.com/canonical/pycloudlib/pull/361 updates the correct image stream source for Azure minimal noble daily images.

## Test Steps
The following integration test will fall over due to inability to call add-apt-repository in minimal azure images because software-properties-common deb isn't installed by default :
```
CLOUD_INIT_CLOUD_INIT_SOURCE=ppa:cloud-init-dev/dailyCLOUD_INIT_OS_IMAGE=noble CLOUD_INIT_PLATFORM=azure test -e integration-tests-jenkins -- tests/integration_tests/datasources/test_azure.py::test_azure_eject
```

## Checklist

- [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [x] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [ ] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/instances.py [hunk 1]
```diff
@@ -211,6 +211,12 @@ def install_proposed_image(self):
 
     def install_ppa(self):
         log.info("Installing PPA")
+        if self.execute("which add-apt-repository").failed:
+            log.info("Installing missing software-properties-common package")
+            assert self.execute("apt update").ok
```
Comment (TheRealFalcon) [inline tests/integration_tests/instances.py:216]: Can we use `self._apt_update()` here instead?

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Just one request inline but otherwise LGTM.

File: tests/integration_tests/instances.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (blackboxsw) [reply tests/integration_tests/instances.py]: Good thought. thx, also I don't think we need the additional self._apt_update after `add-apt-repository ppa:cloud-init-dev/daily`  as focal++ will automatically call apt update after adding the PPA.Bionic and later all automatically call apt-update for us after add-apt-repository is  called.

**Comment by blackboxsw** [comment]
> Can we use self._apt_update() here instead?

 Done and removed the trailing self._apt_update() as it's unnecessary after an add-apt-repository

**Review by TheRealFalcon — APPROVED** [review]

**Pull request merged**